agent/http_oss.go

```diff
@@ -3,10 +3,37 @@
 package agent
 
 import (
+	"fmt"
 	"net/http"
+	"strings"
 
 	"github.com/hashicorp/consul/agent/structs"
 )
 
-func (s *HTTPServer) parseEntMeta(req *http.Request, entMeta *structs.EnterpriseMeta) {
+func (s *HTTPServer) parseEntMeta(req *http.Request, entMeta *structs.EnterpriseMeta) error {
+	if headerNS := req.Header.Get("X-Consul-Namespace"); headerNS != "" {
+		return BadRequestError{Reason: "Invalid header: \"X-Consul-Namespace\" - Namespaces is a Consul Enterprise feature"}
+	}
+	if queryNS := req.URL.Query().Get("ns"); queryNS != "" {
+		return BadRequestError{Reason: "Invalid query parameter: \"ns\" - Namespaces is a Consul Enterprise feature"}
+	}
+	return nil
+}
+
+func (s *HTTPServer) handleUnknownEnterpriseFields(err error) error {
+	if err == nil {
```

Comment:
`err` has been checked before the function was called, which is why I would remove this check.

Reply:
I changed up the invocation of this function so that we now pass in an error var completely unchecked so this branch is still used.